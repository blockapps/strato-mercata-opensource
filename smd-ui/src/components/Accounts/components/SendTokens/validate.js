import { isOauthEnabled } from "../../../../lib/checkMode";

export const validate = (values) => {
  const errors = {};

  if (!values.from) {
    errors.value = 'Please select a user';
  }

  if (!values.fromAddress) {
    errors.value = 'Please select a address';
  }

  if (isOauthEnabled() && !values.address) {
    errors.value = "Please enter address"
  }

  if (!isOauthEnabled()) {

    if (!values.password) {
      errors.value = 'Please enter a password';
    }
    if (!values.radio && !values.toAddress) {
      errors.value = "Please select address"
    }

    if (values.radio === "0" && !values.toAddress) {
      errors.value = "Please select address"
    }

    if (values.radio === "1" && !values.address) {
      errors.value = "Please enter address"
    }
  }

  if (!values.value) {
    errors.value = 'Please enter a value';
  }

  return errors;
};

export default validate;
