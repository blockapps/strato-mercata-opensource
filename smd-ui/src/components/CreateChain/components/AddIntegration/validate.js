export function validate(values) {
  const errors = {};

  return errors;
}