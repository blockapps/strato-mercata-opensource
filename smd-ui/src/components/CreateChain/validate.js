export function validate(values) {
  const errors = {};

  if (!values.chainName) {
    errors.chainName = 'required';
  }

  if (values.members && !values.members.length) {
    errors.members = 'at least add one member';
  }

  return errors;
}