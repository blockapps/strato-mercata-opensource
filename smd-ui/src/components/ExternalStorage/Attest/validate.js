import { isOauthEnabled } from "../../../lib/checkMode";

export default function validate(values) {
  const errors = {};

  if (!isOauthEnabled() && !values.username) {
    errors.username = 'Select username';
  }

  if (!isOauthEnabled() && !values.address) {
    errors.address = 'Address required';
  }

  if (!isOauthEnabled() && !values.password) {
    errors.password = 'Password required';
  }

  if (!values.contractAddress) {
    errors.contractAddress = "Can't be blank";
  }

  return errors;
}