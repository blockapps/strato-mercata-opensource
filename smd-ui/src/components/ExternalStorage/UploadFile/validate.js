import { isOauthEnabled } from "../../../lib/checkMode";

export function validate(values) {

  const errors = {};

  if (!isOauthEnabled() && !values.username) {
    errors.username = 'Select username';
  }

  if (!isOauthEnabled() && !values.address) {
    errors.address = 'Address required';
  }

  if (!isOauthEnabled() && !values.password) {
    errors.password = 'Password required';
  }

  if (!values.content) {
    errors.content = 'Attach file';
  }

  if (!values.provider) {
    errors.provider = 'Select provider';
  }

  if (!values.description) {
    errors.description = 'Attach file';
  }

  return errors;
}