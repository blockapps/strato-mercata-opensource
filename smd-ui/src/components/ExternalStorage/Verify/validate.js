export default function validate(values) {
  const errors = {};

  if (!values.contractAddress) {
    errors.contractAddress = "Can't be blank";
  }

  return errors;
}