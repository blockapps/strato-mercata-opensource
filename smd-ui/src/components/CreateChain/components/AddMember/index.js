import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Button, Dialog, Intent } from '@blueprintjs/core';

import mixpanelWrapper from '../../../../lib/mixpanelWrapper';
import { Field, reduxForm } from 'redux-form';
import { openAddMemberModal, closeAddMemberModal } from '../../createChain.actions';
import { validate } from './validate';

class AddMember extends Component {

  constructor(props) {
    super(props)
    this.state = {
      form: {
      },
      orgName: ``,
      orgUnit: ``,
      commonName: ``,
      access: true,
      errors: null
    }
  }

  closeModal = () => {
    this.props.closeAddMemberModal();
  }

  componentDidMount() {
    mixpanelWrapper.track("add_member_loaded");
  }

  handleOrgNameChange(event) {
    this.setState({
      orgName: event.target.value
    });
  }

  handleOrgUnitChange(event) {
    this.setState({
      orgUnit: event.target.value
    });
  }

  handleCommonNameChange(event) {
    this.setState({
      commonName: event.target.value
    });
  }

  handleAccessChange(event) {
    this.setState({
      access: event.target.value
    });
  }

  errorMessageFor(fieldName) {
    if (this.state.errors && this.state.errors[fieldName]) {
      return this.state.errors[fieldName];
    }
    return null;
  }

  submit = () => {
    let data = {
      orgName: this.state.orgName,
      orgUnit: this.state.orgUnit,
      commonName: this.state.commonName,
      access: this.state.access
    }
    
    let errors = validate(data);
    this.setState({ errors });

    if (!Object.values(errors).length) {
      mixpanelWrapper.track('add_member_submit_click_successful');
      this.props.handler(data);
      this.props.reset();
      this.closeModal();
    }
  }

  render() {
    return (
      <div >
        <Button onClick={() => {
          mixpanelWrapper.track("add_member_open_click");
          this.props.openAddMemberModal();
        }} className="pt-intent-primary pt-icon-add"
          style={{ marginTop: '8px' }}
          text="Add Member" />

        <Dialog
          iconName="add"
          isOpen={this.props.isOpen}
          onClose={this.closeModal}
          title="Add Member"
          className="pt-dark"
        >
          <form>
            <div className="pt-dialog-body">

              {
                // !this.state.form.userSelected &&
                <div className="row">
                  <div className="col-sm-3 text-right">
                    <label className="pt-label smd-pad-4">
                      Org Name:
                  </label>
                  </div>
                  <div className="col-sm-9 smd-pad-4">
                    <Field
                      id="orgName"
                      className="form-width pt-input"
                      placeholder="Org Name"
                      name="orgName"
                      component="input"
                      dir="auto"
                      title="orgName"
                      onChange={(e) => this.handleOrgNameChange(e)}
                      required
                    />
                    <br /><span className="error-text">{this.errorMessageFor('orgName')}</span>
                  </div>
                </div>
              }

              <div className="row">
                <div className="col-sm-3 text-right">
                  <label className="pt-label smd-pad-4">
                    Org Unit:
                  </label>
                </div>
                <div className="col-sm-9 smd-pad-4">
                  <div className="form-width">
                    <Field
                      name="orgUnit"
                      component="input"
                      type="text"
                      placeholder="Org Unit"
                      value={this.state.orgUnit}
                      className="pt-input form-width"
                      onChange={(e) => this.handleOrgUnitChange(e)}
                      required
                    />
                    <br /><span className="error-text">{this.errorMessageFor('orgUnit')}</span>
                  </div>
                </div>
              </div>

              <div className="row">
                <div className="col-sm-3 text-right">
                  <label className="pt-label smd-pad-4">
                    Common Name:
                  </label>
                </div>
                <div className="col-sm-9 smd-pad-4">
                  <div className="form-width">
                    <Field
                      name="commonName"
                      component="input"
                      type="text"
                      placeholder="Common Name"
                      value={this.state.commonName}
                      className="pt-input form-width"
                      onChange={(e) => this.handleCommonNameChange(e)}
                      required
                    />
                    <br /><span className="error-text">{this.errorMessageFor('commonName')}</span>
                  </div>
                </div>
              </div>
              <div className="row">
                <div className="col-sm-3 text-right">
                  <label className="pt-label smd-pad-4">
                    Access
                  </label>
                </div>
                <div className="col-sm-9 smd-pad-4">
                  <div className="form-width">
                  <Field
                      style={{ marginLeft: 25 }}
                      name="radio"
                      component="input"
                      type="radio"
                      value={true}
                      label='Add'
                      checked={this.state.access === true}
                      onClick={() => {
                        this.setState((prevState) => {
                          return {access:true };
                        });
                      }}
                    /> Add
                  <Field
                      style={{ marginLeft: 25 }}
                      name="radio"
                      component="input"
                      type="radio"
                      value={false}
                      label='Remove'
                      checked={this.state.access === false}
                      onClick={() => {
                        this.setState((prevState) => {
                          return {access:false };
                        });
                      }}
                    /> Remove
                    <br /><span className="error-text">{this.errorMessageFor('access')}</span>
                  </div>
                </div>
              </div>
            </div>

            <div className="pt-dialog-footer">
              <div className="pt-dialog-footer-actions">
                <Button text="Cancel" onClick={() => {
                  mixpanelWrapper.track("add_member_cancel");
                  this.props.reset();
                  this.closeModal();
                }} />
                <Button
                  intent={Intent.PRIMARY}
                  onClick={this.submit}
                  text="Add Member"
                />
              </div>
            </div>
          </form>
        </Dialog>
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    isOpen: state.createChain.isAddMemberModalOpen,
    initialValues: {
      orgName: "",
      orgUnit: "",
      commonName: "",
      access: true
    }
  };
}

const formed = reduxForm({ form: 'add-member' })(AddMember);
const connected = connect(mapStateToProps, {
  openAddMemberModal,
  closeAddMemberModal
})(formed);

export default withRouter(connected);
