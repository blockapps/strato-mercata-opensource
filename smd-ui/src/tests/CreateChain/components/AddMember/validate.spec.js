import { validate } from '../../../../components/CreateChain/components/AddMember/validate';

describe('AddMember: validate', () => {

  describe('when form has', () => {

    test('values (with user)', () => {
      const data = {
        username: 'tanuj1000',
        address: 'f11b5c42f5b84efa07f6b0a32c3fc545ff509126',
        enode: 'enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@172.16.0.5:30303?discport=30303',
        balance: 100
      };
      const userSelected = true;

      expect(validate(data, userSelected)).toMatchSnapshot();
    });

    test('values (with address)', () => {
      const data = {
        username: null,
        address: 'f11b5c42f5b84efa07f6b0a32c3fc545ff509126',
        enode: 'enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@172.16.0.5:30303?discport=30303',
        balance: 100
      };
      const userSelected = true;

      expect(validate(data, userSelected)).toMatchSnapshot();
    });

    test('no values', () => {
      const data = {
        username: null,
        address: null,
        enode: null,
        balance: 100
      };
      const userSelected = true;

      expect(validate(data, userSelected)).toMatchSnapshot();
    });

  })

});