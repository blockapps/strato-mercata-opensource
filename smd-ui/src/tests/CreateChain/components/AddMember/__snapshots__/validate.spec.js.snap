// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMember: validate when form has no values 1`] = `Object {}`;

exports[`AddMember: validate when form has values (with address) 1`] = `Object {}`;

exports[`AddMember: validate when form has values (with user) 1`] = `Object {}`;
